lib/stages/HFBAccumulate.cpp

```diff
@@ -241,50 +274,65 @@ void HFBAccumulate::main_thread() {
                 (float)(total_timesamples - total_lost_timesamples) / total_timesamples;
 
             // Normalise data
-            normalise_frame(sum_data, in_buffer_ID);
+            normalise_frame(sum_data, in_frame_id);
 
             // Only output integration if there are enough good samples
             if (good_samples_frac >= _good_samples_threshold) {
 
+                auto frame = HFBFrameView(out_buf, out_frame_id);
+
                 // Create new metadata
-                allocate_new_metadata_object(out_buf, out_buffer_ID);
+                allocate_new_metadata_object(out_buf, out_frame_id);
 
-                // Populate metadata
+                // Populate metadata using HFBFrameView
                 int64_t fpga_seq =
                     fpga_seq_num_end_old - ((_num_frames_to_integrate - 1) * _samples_per_data_set);
-                set_fpga_seq_start_hfb(out_buf, out_buffer_ID, fpga_seq);
 
-                // Set GPS time
-                set_ctime_hfb(out_buf, out_buffer_ID, tel.to_time(fpga_seq));
+                frame.fpga_seq_start = fpga_seq;
+                frame.time = tel.to_time(fpga_seq);
+                frame.fpga_seq_total = total_timesamples - total_lost_timesamples;
+                frame.fpga_seq_length = total_timesamples;
+                frame.freq_id = tel.to_freq_id(in_buf, in_frame_id);
+                frame.dataset_id = ds_id;
```

Comment:
Note that after Rick's 2nd stage RFI changes the incoming data will have a dataset_id which needs to be used. I couldn't put a comment in the relevant in the place, but this means that the code which generates the dataset_id needs to be deferred until it receives data and can actually use that as the base for the new dataset id.

Reply:
I think I have addressed this in my latest commit.